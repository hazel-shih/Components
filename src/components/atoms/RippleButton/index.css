.wrapper {
	display: flex;
	justify-content: center;
	align-items: center;
	height: 100vh;
}

.ripple-button {
	position: relative;
	background-color: var(--primary);
	color: white;
	padding: 20px 30px;
	font-weight: 500;
	overflow: hidden;

	& > .circle-effect {
		position: absolute;
		background-color: white;
		width: 100px;
		height: 100px;
		border-radius: 50%;
		transform: translate(-50%, -50%) scale(0.5);
		animation: scale 0.5s ease-in-out;
	}
}

@keyframes scale {
	to {
		transform: translate(-50%, -50%) scale(3);
		opacity: 0;
	}
}
