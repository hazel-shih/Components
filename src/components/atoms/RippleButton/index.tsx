import React, { useState } from 'react';
import classnames from 'classnames';

import styles from './index.css';

interface RippleButtonProperty {
	className?: string;
}

type PointPosition = {
	x: null | number;
	y: null | number;
};

const RippleButton: React.FC<RippleButtonProperty> = ({ className }) => {
	const [pointPosition, setPointPosition] = useState<PointPosition>({
		x: null,
		y: null,
	});

	return (
		<div className={styles.wrapper}>
			<button
				type="button"
				className={classnames(styles.rippleButton, className)}
				onClick={(e: React.MouseEvent) => {
					const { clientX, clientY } = e;
					const { offsetTop, offsetLeft } = e.target;
					setPointPosition({
						x: clientX - offsetLeft,
						y: clientY - offsetTop,
					});
					setTimeout(() => {
						setPointPosition({
							x: null,
							y: null,
						});
					}, 500);
				}}
			>
				CLICK ME
				{pointPosition.x && (
					<span
						className={styles.circleEffect}
						style={{ top: `${pointPosition.y}px`, left: `${pointPosition.x}px` }}
					/>
				)}
			</button>
		</div>
	);
};

export default RippleButton;
