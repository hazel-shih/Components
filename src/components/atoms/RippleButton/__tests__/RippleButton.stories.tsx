import React from 'react';
import { storiesOf } from '@storybook/react';

import RippleButton from 'components/atoms/RippleButton';

const stories = storiesOf('atoms/RippleButton', module);

stories.add('__interactive', () => <RippleButton />);
