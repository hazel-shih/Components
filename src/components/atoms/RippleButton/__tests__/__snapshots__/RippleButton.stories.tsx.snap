// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <button
      class="rippleButton"
      type="button"
    >
      CLICK ME
    </button>
  </div>
</DocumentFragment>
`;
